# Data Collection

In [ ]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("prasad22/healthcare-dataset")

print("Path to dataset files:", path)

Using Colab cache for faster access to the 'healthcare-dataset' dataset.
Path to dataset files: /kaggle/input/healthcare-dataset


In [ ]:
import pandas as pd
import os

# List files in the downloaded directory to find the correct CSV
file_list = os.listdir(path)
print("Files in the dataset directory:", file_list)

# Assuming the CSV file is named 'healthcare_dataset.csv'
csv_file_path = os.path.join(path, 'healthcare_dataset.csv')

df = pd.read_csv(csv_file_path)

print("Dataset loaded successfully! Shape:", df.shape)
df.head()

Files in the dataset directory: ['healthcare_dataset.csv']
Dataset loaded successfully! Shape: (55500, 15)


Name  Age  Gender Blood Type Medical Condition Date of Admission  \
0  Bobby JacksOn   30    Male         B-            Cancer        2024-01-31   
1   LesLie TErRy   62    Male         A+           Obesity        2019-08-20   
2    DaNnY sMitH   76  Female         A-           Obesity        2022-09-22   
3   andrEw waTtS   28  Female         O+          Diabetes        2020-11-18   
4  adrIENNE bEll   43  Female        AB+            Cancer        2022-09-19   

             Doctor                    Hospital Insurance Provider  \
0     Matthew Smith             Sons and Miller         Blue Cross   
1   Samantha Davies                     Kim Inc           Medicare   
2  Tiffany Mitchell                    Cook PLC              Aetna   
3       Kevin Wells  Hernandez Rogers and Vang,           Medicare   
4    Kathleen Hanna                 White-White              Aetna   

   Billing Amount  Room Number Admission Type Discharge Date   Medication  \
0    18856.281306          328         Urgent     2024-02-02  Paracetamol   
1    33643.327287          265      Emergency     2019-08-26    Ibuprofen   
2    27955.096079          205      Emergency     2022-10-07      Aspirin   
3    37909.782410          450       Elective     2020-12-18    Ibuprofen   
4    14238.317814          458         Urgent     2022-10-09   Penicillin   

   Test Results  
0        Normal  
1  Inconclusive  
2        Normal  
3      Abnormal  
4      Abnormal

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55500 entries, 0 to 55499
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Name                55500 non-null  object 
 1   Age                 55500 non-null  int64  
 2   Gender              55500 non-null  object 
 3   Blood Type          55500 non-null  object 
 4   Medical Condition   55500 non-null  object 
 5   Date of Admission   55500 non-null  object 
 6   Doctor              55500 non-null  object 
 7   Hospital            55500 non-null  object 
 8   Insurance Provider  55500 non-null  object 
 9   Billing Amount      55500 non-null  float64
 10  Room Number         55500 non-null  int64  
 11  Admission Type      55500 non-null  object 
 12  Discharge Date      55500 non-null  object 
 13  Medication          55500 non-null  object 
 14  Test Results        55500 non-null  object 
dtypes: float64(1), int64(2), object(12)
memory usage: 6.4

In [ ]:
df.describe()

Age  Billing Amount   Room Number
count  55500.000000    55500.000000  55500.000000
mean      51.539459    25539.316097    301.134829
std       19.602454    14211.454431    115.243069
min       13.000000    -2008.492140    101.000000
25%       35.000000    13241.224652    202.000000
50%       52.000000    25538.069376    302.000000
75%       68.000000    37820.508436    401.000000
max       89.000000    52764.276736    500.000000

In [ ]:
df.describe(include='object')

Name Gender Blood Type Medical Condition Date of Admission  \
count         55500  55500      55500             55500             55500   
unique        49992      2          8                 6              1827   
top     DAvId muNoZ   Male         A-         Arthritis        2024-03-16   
freq              3  27774       6969              9308                50   

               Doctor   Hospital Insurance Provider Admission Type  \
count           55500      55500              55500          55500   
unique          40341      39876                  5              3   
top     Michael Smith  LLC Smith              Cigna       Elective   
freq               27         44              11249          18655   

       Discharge Date Medication Test Results  
count           55500      55500        55500  
unique           1856          5            3  
top        2020-03-15    Lipitor     Abnormal  
freq               53      11140        18627

# Data Preparation

## Standardize Categorical Features (Lowercase Conversion)

### Subtask:
Convert all categorical columns to lowercase

Reasoning: To ensure consistency across the dataset

In [ ]:
# Identify categorical columns (object dtype) excluding those we've already handled or dropped
categorical_cols = df.select_dtypes(include='object').columns.tolist()

# Convert all identified categorical columns to lowercase
for col in categorical_cols:
    df[col] = df[col].str.lower()

print("DataFrame with all categorical columns converted to lowercase:")
display(df.head())

DataFrame with all categorical columns converted to lowercase:


Name  Age  Gender Blood Type Medical Condition Date of Admission  \
0  bobby jackson   30    male         b-            cancer        2024-01-31   
1   leslie terry   62    male         a+           obesity        2019-08-20   
2    danny smith   76  female         a-           obesity        2022-09-22   
3   andrew watts   28  female         o+          diabetes        2020-11-18   
4  adrienne bell   43  female        ab+            cancer        2022-09-19   

             Doctor                    Hospital Insurance Provider  \
0     matthew smith             sons and miller         blue cross   
1   samantha davies                     kim inc           medicare   
2  tiffany mitchell                    cook plc              aetna   
3       kevin wells  hernandez rogers and vang,           medicare   
4    kathleen hanna                 white-white              aetna   

   Billing Amount  Room Number Admission Type Discharge Date   Medication  \
0    18856.281306          328         urgent     2024-02-02  paracetamol   
1    33643.327287          265      emergency     2019-08-26    ibuprofen   
2    27955.096079          205      emergency     2022-10-07      aspirin   
3    37909.782410          450       elective     2020-12-18    ibuprofen   
4    14238.317814          458         urgent     2022-10-09   penicillin   

   Test Results  
0        normal  
1  inconclusive  
2        normal  
3      abnormal  
4      abnormal

In [ ]:
df.describe(include='object')

Name Gender Blood Type Medical Condition  \
count              55500  55500      55500             55500   
unique             40235      2          8                 6   
top     michael williams   male         a-         arthritis   
freq                  24  27774       6969              9308   

       Date of Admission         Doctor   Hospital Insurance Provider  \
count              55500          55500      55500              55500   
unique              1827          40341      39876                  5   
top           2024-03-16  michael smith  llc smith              cigna   
freq                  50             27         44              11249   

       Admission Type Discharge Date Medication Test Results  
count           55500          55500      55500        55500  
unique              3           1856          5            3  
top          elective     2020-03-15    lipitor     abnormal  
freq            18655             53      11140        18627

## Handle missing values

### Subtask:
Check for and address any missing values in the dataset.


**Reasoning**:
Check for missing values in the DataFrame.



In [ ]:
print("Missing values per column:")
print(df.isnull().sum())

Missing values per column:
Name                  0
Age                   0
Gender                0
Blood Type            0
Medical Condition     0
Date of Admission     0
Doctor                0
Hospital              0
Insurance Provider    0
Billing Amount        0
Room Number           0
Admission Type        0
Discharge Date        0
Medication            0
Test Results          0
dtype: int64


## Convert data types

### Subtask:
Convert columns with incorrect data types, such as converting date columns to datetime objects.


**Reasoning**:
Identify and convert date columns to datetime objects.



In [ ]:
date_columns = ['Date of Admission', 'Discharge Date']
for col in date_columns:
  df[col] = pd.to_datetime(df[col])

print("Data types after conversion:")
display(df.info())

Data types after conversion:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55500 entries, 0 to 55499
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Name                55500 non-null  object        
 1   Age                 55500 non-null  int64         
 2   Gender              55500 non-null  object        
 3   Blood Type          55500 non-null  object        
 4   Medical Condition   55500 non-null  object        
 5   Date of Admission   55500 non-null  datetime64[ns]
 6   Doctor              55500 non-null  object        
 7   Hospital            55500 non-null  object        
 8   Insurance Provider  55500 non-null  object        
 9   Billing Amount      55500 non-null  float64       
 10  Room Number         55500 non-null  int64         
 11  Admission Type      55500 non-null  object        
 12  Discharge Date      55500 non-null  datetime64[ns]
 13  Medication       

None

## Address inconsistencies

### Subtask:
Address inconsistencies identified during data exploration, like standardizing categorical entries or addressing potential misspellings if necessary.


**Reasoning**:
Examine the unique values and their counts for the specified categorical columns to identify inconsistencies.



In [ ]:
categorical_cols = ['Gender', 'Blood Type', 'Medical Condition', 'Insurance Provider', 'Admission Type', 'Test Results']

for col in categorical_cols:
  print(f"\nUnique values and counts for {col}:")
  display(df[col].value_counts())


Unique values and counts for Gender:


Gender
male      27774
female    27726
Name: count, dtype: int64


Unique values and counts for Blood Type:


Blood Type
a-     6969
a+     6956
ab+    6947
ab-    6945
b+     6945
b-     6944
o+     6917
o-     6877
Name: count, dtype: int64


Unique values and counts for Medical Condition:


Medical Condition
arthritis       9308
diabetes        9304
hypertension    9245
obesity         9231
cancer          9227
asthma          9185
Name: count, dtype: int64


Unique values and counts for Insurance Provider:


Insurance Provider
cigna               11249
medicare            11154
unitedhealthcare    11125
blue cross          11059
aetna               10913
Name: count, dtype: int64


Unique values and counts for Admission Type:


Admission Type
elective     18655
urgent       18576
emergency    18269
Name: count, dtype: int64


Unique values and counts for Test Results:


Test Results
abnormal        18627
normal          18517
inconclusive    18356
Name: count, dtype: int64

## Addressing High Cardinality

### Subtask: Drop 'Name' Column

**Reasoning**:
Remove the 'Name' column as it is not needed for analysis.

In [ ]:
df = df.drop('Name', axis=1)

print("DataFrame after dropping 'Name' column:")
display(df.head())

DataFrame after dropping 'Name' column:


Age  Gender Blood Type Medical Condition Date of Admission  \
0   30    male         b-            cancer        2024-01-31   
1   62    male         a+           obesity        2019-08-20   
2   76  female         a-           obesity        2022-09-22   
3   28  female         o+          diabetes        2020-11-18   
4   43  female        ab+            cancer        2022-09-19   

             Doctor                    Hospital Insurance Provider  \
0     matthew smith             sons and miller         blue cross   
1   samantha davies                     kim inc           medicare   
2  tiffany mitchell                    cook plc              aetna   
3       kevin wells  hernandez rogers and vang,           medicare   
4    kathleen hanna                 white-white              aetna   

   Billing Amount  Room Number Admission Type Discharge Date   Medication  \
0    18856.281306          328         urgent     2024-02-02  paracetamol   
1    33643.327287          265      emergency     2019-08-26    ibuprofen   
2    27955.096079          205      emergency     2022-10-07      aspirin   
3    37909.782410          450       elective     2020-12-18    ibuprofen   
4    14238.317814          458         urgent     2022-10-09   penicillin   

   Test Results  
0        normal  
1  inconclusive  
2        normal  
3      abnormal  
4      abnormal

### Subtask: Discuss Strategies for High Cardinality Columns ('Doctor' and 'Hospital')

**Reasoning**:
Provide potential strategies for handling the high number of unique values in the 'Doctor' and 'Hospital' columns.

In [ ]:
df['doctor_freq'] = df['Doctor'].map(df['Doctor'].value_counts())
df['hospital_freq'] = df['Hospital'].map(df['Hospital'].value_counts())

print("DataFrame with Doctor and Hospital frequencies:")
display(df[['Doctor', 'doctor_freq', 'Hospital', 'hospital_freq']].head())

DataFrame with Doctor and Hospital frequencies:


Doctor  doctor_freq                    Hospital  hospital_freq
0     matthew smith           17             sons and miller              7
1   samantha davies            1                     kim inc              6
2  tiffany mitchell            1                    cook plc              4
3       kevin wells            1  hernandez rogers and vang,              1
4    kathleen hanna            2                 white-white              2

## Feature Engineering

### Subtask: Extract Year and Day of Week from Dates

**Reasoning**:
Extract the year and day of the week from the 'Date of Admission' and 'Discharge Date' columns to analyze temporal patterns.

In [ ]:
df['Admission_Year'] = df['Date of Admission'].dt.year
df['Discharge_Year'] = df['Discharge Date'].dt.year
df['Admission_DayOfWeek'] = df['Date of Admission'].dt.day_name()
df['Discharge_DayOfWeek'] = df['Discharge Date'].dt.day_name()
df['Admission_Weekday'] = df['Date of Admission'].dt.weekday < 5 # Monday=0, Sunday=6
df['Discharge_Weekday'] = df['Discharge Date'].dt.weekday < 5

print("DataFrame with new date features:")
display(df[['Date of Admission', 'Admission_Year', 'Admission_DayOfWeek', 'Admission_Weekday', 'Discharge Date', 'Discharge_Year', 'Discharge_DayOfWeek', 'Discharge_Weekday']].head())

DataFrame with new date features:


Date of Admission  Admission_Year Admission_DayOfWeek  Admission_Weekday  \
0        2024-01-31            2024           Wednesday               True   
1        2019-08-20            2019             Tuesday               True   
2        2022-09-22            2022            Thursday               True   
3        2020-11-18            2020           Wednesday               True   
4        2022-09-19            2022              Monday               True   

  Discharge Date  Discharge_Year Discharge_DayOfWeek  Discharge_Weekday  
0     2024-02-02            2024              Friday               True  
1     2019-08-26            2019              Monday               True  
2     2022-10-07            2022              Friday               True  
3     2020-12-18            2020              Friday               True  
4     2022-10-09            2022              Sunday              False

### Subtask: Calculate Admission Duration

**Reasoning**:
Calculate the duration of each patient's admission by subtracting the 'Date of Admission' from the 'Discharge Date'.

In [ ]:
df['Admission_Duration'] = (df['Discharge Date'] - df['Date of Admission']).dt.days

print("DataFrame with Admission Duration:")
display(df[['Date of Admission', 'Discharge Date', 'Admission_Duration']].head())
display(df['Admission_Duration'].describe())

DataFrame with Admission Duration:


Date of Admission Discharge Date  Admission_Duration
0        2024-01-31     2024-02-02                   2
1        2019-08-20     2019-08-26                   6
2        2022-09-22     2022-10-07                  15
3        2020-11-18     2020-12-18                  30
4        2022-09-19     2022-10-09                  20

count    55500.000000
mean        15.509009
std          8.659600
min          1.000000
25%          8.000000
50%         15.000000
75%         23.000000
max         30.000000
Name: Admission_Duration, dtype: float64

### Subtask: Create Age Bins

**Reasoning**:
Bin the 'Age' into 5 categories to simplify age-based analysis.

In [ ]:
df['Age_Bin'] = pd.cut(df['Age'], bins=5, labels=['13-27', '28-42', '43-57', '58-73', '74-89'])

print("DataFrame with Age Bins:")
display(df[['Age', 'Age_Bin']].head())
display(df['Age_Bin'].value_counts())

DataFrame with Age Bins:


Age Age_Bin
0   30   28-42
1   62   58-73
2   76   74-89
3   28   13-27
4   43   28-42

Age_Bin
43-57    12442
28-42    12195
58-73    12184
74-89     9773
13-27     8906
Name: count, dtype: int64

### Subtask: Create Age Bins (Teen, Young Adult, Adult, Senior)

**Reasoning**:
Re-display the results of the age binning with 'Teen', 'Young Adult', 'Adult', and 'Senior' categories.

In [ ]:
# Define age bins and labels based on the provided categories and age range (13-89)
# Bin 1: Teen (13-19)
# Bin 2: Young Adult (20-39)
# Bin 3: Adult (40-59)
# Bin 4: Senior (60-79)
# Bin 5: Elderly Senior (80-89) - grouping the upper age range

bins = [12, 19, 39, 59, 79, 90] # Adjusted bins to include 13 and exclude 90 as upper limit
labels = ['Teen', 'Young Adult', 'Adult', 'Senior', 'Elderly Senior'] # Adjusted labels to reflect the bins

df['Age_Bin'] = pd.cut(df['Age'], bins=bins, labels=labels, right=False) # Use right=False to include the left edge

print("DataFrame with updated Age Bins (Teen, Young Adult, Adult, Senior, Elderly Senior):")
display(df[['Age', 'Age_Bin']].head())
display(df['Age_Bin'].value_counts())

DataFrame with updated Age Bins (Teen, Young Adult, Adult, Senior, Elderly Senior):


Age      Age_Bin
0   30  Young Adult
1   62       Senior
2   76       Senior
3   28  Young Adult
4   43        Adult

Age_Bin
Adult             16422
Young Adult       16233
Senior            16222
Elderly Senior     5735
Teen                888
Name: count, dtype: int64

### Subtask: Categorize Billing Amount

**Reasoning**:
Categorize 'Billing Amount' into 'Low', 'Medium', and 'High' based on quantiles to analyze billing patterns.

In [ ]:
df['Billing_Range'] = pd.qcut(df['Billing Amount'], q=3, labels=['Low', 'Medium', 'High'])

print("DataFrame with Billing Range:")
display(df[['Billing Amount', 'Billing_Range']].head())
display(df['Billing_Range'].value_counts())

DataFrame with Billing Range:


Billing Amount Billing_Range
0    18856.281306        Medium
1    33643.327287        Medium
2    27955.096079        Medium
3    37909.782410          High
4    14238.317814           Low

Billing_Range
Low       18500
Medium    18500
High      18500
Name: count, dtype: int64

### Subtask: Extract Floor from Room Number

**Reasoning**:
Extract the floor number from the 'Room Number' assuming room numbers indicate the floor (e.g., room 301 is on the 3rd floor).

In [ ]:
df['Floor'] = df['Room Number'].astype(str).str[0]

print("DataFrame with Floor:")
display(df[['Room Number', 'Floor']].head())
display(df['Floor'].value_counts())

DataFrame with Floor:


Room Number Floor
0          328     3
1          265     2
2          205     2
3          450     4
4          458     4

Floor
3    14112
4    13883
2    13834
1    13544
5      127
Name: count, dtype: int64

### Subtask: Bin Doctor and Hospital Frequencies

**Reasoning**:
Bin the doctor and hospital frequencies to categorize them based on their occurrence in the dataset.

In [ ]:
# Bin doctor frequencies into categories (e.g., Low, Medium, High frequency)
# We can use quantiles to create bins with roughly equal numbers of observations
df['doctor_freq_bin'] = pd.qcut(df['doctor_freq'], q=4, duplicates='drop')

# Bin hospital frequencies into categories
df['hospital_freq_bin'] = pd.qcut(df['hospital_freq'], q=4, duplicates='drop')

print("DataFrame with Doctor and Hospital Frequency Bins:")
display(df[['Doctor', 'doctor_freq', 'doctor_freq_bin', 'Hospital', 'hospital_freq', 'hospital_freq_bin']].head())

print("\nValue counts for Doctor Frequency Bins:")
display(df['doctor_freq_bin'].value_counts())

print("\nValue counts for Hospital Frequency Bins:")
display(df['hospital_freq_bin'].value_counts())

DataFrame with Doctor and Hospital Frequency Bins:


Doctor  doctor_freq doctor_freq_bin                    Hospital  \
0     matthew smith           17     (2.0, 27.0]             sons and miller   
1   samantha davies            1    (0.999, 2.0]                     kim inc   
2  tiffany mitchell            1    (0.999, 2.0]                    cook plc   
3       kevin wells            1    (0.999, 2.0]  hernandez rogers and vang,   
4    kathleen hanna            2    (0.999, 2.0]                 white-white   

   hospital_freq hospital_freq_bin  
0              7       (2.0, 44.0]  
1              6       (2.0, 44.0]  
2              4       (2.0, 44.0]  
3              1      (0.999, 2.0]  
4              2      (0.999, 2.0]


Value counts for Doctor Frequency Bins:


doctor_freq_bin
(0.999, 2.0]    44756
(2.0, 27.0]     10744
Name: count, dtype: int64


Value counts for Hospital Frequency Bins:


hospital_freq_bin
(0.999, 2.0]    43296
(2.0, 44.0]     12204
Name: count, dtype: int64

In [ ]:
print("Min and Max Doctor Frequency:", df['doctor_freq'].min(), df['doctor_freq'].max())
print("Min and Max Hospital Frequency:", df['hospital_freq'].min(), df['hospital_freq'].max())

Min and Max Doctor Frequency: 1 27
Min and Max Hospital Frequency: 1 44


### Subtask: One-Hot Encode Hospital Frequency Bins

**Reasoning**:
One-hot encode the 'hospital_freq_bin' column to convert the categorical frequency ranges into numerical features.

In [ ]:
df = pd.get_dummies(df, columns=['hospital_freq_bin'], prefix='hospital_freq')
df = pd.get_dummies(df, columns=['doctor_freq_bin'], prefix='doctor_freq')

print("DataFrame after One-Hot Encoding Hospital Frequency Bins:")
display(df.head())
display(df.columns)

DataFrame after One-Hot Encoding Hospital Frequency Bins:


Age  Gender Blood Type Medical Condition Date of Admission  \
0   30    male         b-            cancer        2024-01-31   
1   62    male         a+           obesity        2019-08-20   
2   76  female         a-           obesity        2022-09-22   
3   28  female         o+          diabetes        2020-11-18   
4   43  female        ab+            cancer        2022-09-19   

             Doctor                    Hospital Insurance Provider  \
0     matthew smith             sons and miller         blue cross   
1   samantha davies                     kim inc           medicare   
2  tiffany mitchell                    cook plc              aetna   
3       kevin wells  hernandez rogers and vang,           medicare   
4    kathleen hanna                 white-white              aetna   

   Billing Amount  Room Number  ... Admission_Weekday Discharge_Weekday  \
0    18856.281306          328  ...              True              True   
1    33643.327287          265  ...              True              True   
2    27955.096079          205  ...              True              True   
3    37909.782410          450  ...              True              True   
4    14238.317814          458  ...              True             False   

  Admission_Duration      Age_Bin  Billing_Range  Floor  \
0                  2  Young Adult         Medium      3   
1                  6       Senior         Medium      2   
2                 15       Senior         Medium      2   
3                 30  Young Adult           High      4   
4                 20        Adult            Low      4   

   hospital_freq_(0.999, 2.0]  hospital_freq_(2.0, 44.0]  \
0                       False                       True   
1                       False                       True   
2                       False                       True   
3                        True                      False   
4                        True                      False   

  doctor_freq_(0.999, 2.0] doctor_freq_(2.0, 27.0]  
0                    False                    True  
1                     True                   False  
2                     True                   False  
3                     True                   False  
4                     True                   False  

[5 rows x 30 columns]

Index(['Age', 'Gender', 'Blood Type', 'Medical Condition', 'Date of Admission',
       'Doctor', 'Hospital', 'Insurance Provider', 'Billing Amount',
       'Room Number', 'Admission Type', 'Discharge Date', 'Medication',
       'Test Results', 'doctor_freq', 'hospital_freq', 'Admission_Year',
       'Discharge_Year', 'Admission_DayOfWeek', 'Discharge_DayOfWeek',
       'Admission_Weekday', 'Discharge_Weekday', 'Admission_Duration',
       'Age_Bin', 'Billing_Range', 'Floor', 'hospital_freq_(0.999, 2.0]',
       'hospital_freq_(2.0, 44.0]', 'doctor_freq_(0.999, 2.0]',
       'doctor_freq_(2.0, 27.0]'],
      dtype='object')

### Subtask: Create Feature for Doctors in Multiple Hospitals

**Reasoning**:
Create a binary feature indicating whether a doctor appeared in more than one hospital.

In [ ]:
# Group by Doctor and count the number of unique hospitals
doctor_hospital_counts = df.groupby('Doctor')['Hospital'].nunique()

# Filter for doctors who appeared in more than one hospital
doctors_in_multiple_hospitals = doctor_hospital_counts[doctor_hospital_counts > 1]

print("Doctors who appeared in more than one hospital:")
display(doctors_in_multiple_hospitals)

Doctors who appeared in more than one hospital:


Doctor
aaron baker         2
aaron brown         6
aaron collins       2
aaron cook          2
aaron curtis        2
                   ..
zachary sanchez     2
zachary smith       5
zachary thomas      2
zachary thompson    2
zachary wright      2
Name: Hospital, Length: 5951, dtype: int64

In [ ]:
# Create a list of doctors who appeared in more than one hospital
doctors_in_multiple = doctors_in_multiple_hospitals.index.tolist()

# Create a new feature in the main DataFrame
df['doctor_in_multiple_hospitals'] = df['Doctor'].apply(lambda x: x in doctors_in_multiple)

print("DataFrame with 'doctor_in_multiple_hospitals' feature:")
display(df[['Doctor', 'doctor_in_multiple_hospitals']].head())
display(df['doctor_in_multiple_hospitals'].value_counts())

DataFrame with 'doctor_in_multiple_hospitals' feature:


Doctor  doctor_in_multiple_hospitals
0     matthew smith                          True
1   samantha davies                         False
2  tiffany mitchell                         False
3       kevin wells                         False
4    kathleen hanna                         False

doctor_in_multiple_hospitals
False    38204
True     17296
Name: count, dtype: int64

### Subtask: Save Processed DataFrame to CSV

**Reasoning**:
Save the processed DataFrame to a CSV file for future use or analysis.

In [ ]:
output_csv_path = 'healthcare_dataset_processed.csv'
df.to_csv(output_csv_path, index=False)

print(f"Processed DataFrame saved to {output_csv_path}")

Processed DataFrame saved to healthcare_dataset_processed.csv
